internal/util-interface/src/main/java/xsbti/ScalaAction.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * sbt
+ * Copyright 2011 - 2018, Lightbend, Inc.
+ * Copyright 2008 - 2010, Mark Harrah
+ * Licensed under Apache License 2.0 (see LICENSE)
+ */
+
+package xsbti;
+
+import java.util.Optional;
+
+/**
+ * A ScalaAction is very miminal representation of a `CodeAction` in the LSP protocol.
+ *
+ * <p>However it only focuses on the actual title, description, and edit, leaving it up to the
+ * language server to communicate with the client and put together a proper codeAction in accordance
+ * to client capabilities.
+ *
+ * @see <a href=
+ *     "https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#codeAction">`CodeAction`</a>
+ */
+public interface ScalaAction {
```

Comment:
I wonder if this should be `Action` instead of `ScalaAction`. Who knows, maybe one day we can have actions coming from Java.

Or from a code generator: using the sourcePositionMappers, sbt would be able to transform the Scala actions into Twirl actions for instance.

Reply:
Ooooo, interesting idea. Yea, I can change this to `Action`.